### PR TITLE
feat: Dashboard > Home: Card expenses section display exact amount

### PR DESCRIPTION
**Pull request by SahilK-027**
## Clickup
[clickup link](https://app.clickup.com/t/86cx5qj9j)

## UI Preview

https://github.com/user-attachments/assets/526a3f50-5e34-4026-be42-271a34843bd9

## Summary by CodeRabbit

- **New Features**
  - Enhanced display logic for total draft and amount values based on screen size.

- **Style**
  - Updated typography for various classes in the Card Detail component for improved visual consistency.

- **Tests**
  - Added `ExactCurrencyPipe` to the test suite for enhanced testing capabilities of the Card Detail component. 

- **Chores**
  - Introduced a new property to manage screen size conditions in the Card Detail component.

### CODE REVIEW

**Review by Dimple16 — CHANGES_REQUESTED** [review]
same comments as https://github.com/fylein/fyle-mobile-app/pull/3365#pullrequestreview-2508464092

File: src/app/shared/components/spent-cards/card-detail/card-detail.component.ts [hunk 1]
```diff
@@ -9,13 +9,16 @@ import { TrackingService } from 'src/app/core/services/tracking.service';
...
-export class CardDetailComponent {
+export class CardDetailComponent implements OnInit {
   @Input() cardDetail: PlatformCorporateCardDetail;
 
   @Input() homeCurrency: string;
 
   @Input() currencySymbol: string;
 
+  // To track if the screen is small (320px or below)
+  isSmallScreen = false;
```
Comment (Dimple16) [inline src/app/shared/components/spent-cards/card-detail/card-detail.component.ts:20]: same comment as previous PR

File: src/app/shared/components/spent-cards/card-detail/card-detail.component.ts [hunk 2]
```diff
@@ -9,13 +9,16 @@ import { TrackingService } from 'src/app/core/services/tracking.service';
   templateUrl: './card-detail.component.html',
   styleUrls: ['./card-detail.component.scss'],
 })
-export class CardDetailComponent {
+export class CardDetailComponent implements OnInit {
```
Comment (Dimple16) [inline src/app/shared/components/spent-cards/card-detail/card-detail.component.ts:12]: can remove

File: src/app/shared/components/spent-cards/card-detail/card-detail.component.html [hunk 3]
```diff
@@ -25,7 +25,16 @@
           <span class="stats--expenses-amount-currency">
             {{ currencySymbol }}
           </span>
-          {{ cardDetail.stats.totalDraftValue | humanizeCurrency : homeCurrency : true }}
+          <ng-container
+            *ngIf="!isSmallScreen && cardDetail.stats.totalDraftValue < 1000000000; else humanizeDraftAmount"
```
Comment (Dimple16) [inline src/app/shared/components/spent-cards/card-detail/card-detail.component.html:29]: how do we handle `-ve` amount cases?

**Review by Dimple16 — CHANGES_REQUESTED** [review]
added comments

File: src/app/shared/components/spent-cards/card-detail/card-detail.component.html (hunk 3 above)
Comment (SahilK-027) [reply src/app/shared/components/spent-cards/card-detail/card-detail.component.html:29]: Ahh, I missed this case, thanks for pointing it out, I will add a better check here. 👍🏻

Comment (SahilK-027) [reply src/app/shared/components/spent-cards/card-detail/card-detail.component.html]: Added check to display only 9digits of -ve amount

**Review by Dimple16 — APPROVED** [review]

**Pull request merged**